src/components/Form/Input/CurrencyInput.tsx

```diff
@@ -1,31 +1,213 @@
-import { useState, useCallback, useEffect } from 'react'
-import { getValueWithDecimalFormat, unFormatCurrency } from 'dd360-utils'
-import BaseInput, { InputProps } from './BaseInput'
-
-function CurrencyInput(props: InputProps) {
-  const { onChange, value, prefix = '' } = props
-  const [localValue, setLocalValue] = useState(
-    getValueWithDecimalFormat(String(value))
-  )
-
-  const handleChange = useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>) => {
-      event.target.value = getValueWithDecimalFormat(event.target.value)
-      setLocalValue(event.target.value)
-      event.target.value = unFormatCurrency(event.target.value).toString()
-      onChange && onChange(event)
-    },
-    [onChange, localValue]
-  )
-
-  useEffect(() => {
-    if (!isNaN(Number(value)))
-      setLocalValue(getValueWithDecimalFormat(String(value)))
-  }, [value])
-
-  return (
-    <BaseInput {...props} onChange={handleChange} value={prefix + localValue} />
-  )
+import {
+  useState,
+  useCallback,
+  InputHTMLAttributes,
+  ReactNode,
+  useRef,
+  forwardRef
+} from 'react'
+import CInput from 'react-currency-input-field'
+import { composeClasses } from 'lib/classes'
+import {
+  inputVariants,
+  InputVariant as InputVariantType,
+  InputType,
+  getPaddingInput
+} from '../shared'
+import { Padding, Rounded, ShadowVariants } from '../../../interfaces/types'
+import FormLabel from '../FormLabel'
+import { IconStatus } from './BaseInput'
+import useInputStyles from './useInputStyles'
+
+interface InputCurrencyProps extends InputHTMLAttributes<HTMLInputElement> {
```

Comment:
some props are exactly equal than InputProps, we can import this interface and extends, to avoid repeat code